### DISCUSSION TITLE
Assistance with TypeScript helper function for setting errors

### DISCUSSION THREAD

**Discussion by jamiehaywood** [react-hook-form/react-hook-form#3354] [Q&A]
Hi RHF community!

I'm trying to create a helper function that will set form errors when they get returned from our API, but I'm struggling to get the types working.

My general idea is:
```
let { errors } = await userLogin();

for (const error in errors) {
  setError(error as Extract<keyof LoginFormInputs, string>, {
    type: 'server',
    message: errors[error],
  });
}
```

but making it generic so that I can reuse a `setAllTheErrors` function on every page by just passing in my server `errors` object & the RHF `setError` function as a callback.

I've got to this stage (using the RHF types), but I can't for the life of me get it working / understand what I'm doing wrong.

```
function setAllTheErrors<TFieldValues extends FieldValues = FieldValues>(
  setError: (name: FieldName<TFieldValues>, error: ErrorOption) => void,
  errors: Record<string, string>
) {
  for (const error in errors) {
    setError(error as Extract<keyof TFieldValues, string>, {
      type: 'server',
      message: errors[error],
    });
  }
}
```

Thanks for taking the time to read 🙂

**Comment by skiirx** [comment]
@jamiehaywood Hi! Did you made a success over this thing? 🙂

**Reply by jamiehaywood** [reply]
hi @skiirx, I didn't unfortunately. It was in an old work project - I did manage to get something similar working. When I've got a moment, I'll take a look at the code and post back :)
